### PR TITLE
ServiceInstaller updates

### PR DESCRIPTION
**Pull request by jakebrown58**
ServiceInstaller now copies files from the zip archive to program files.
Better logging and exit codes.
Better detection of failures.

### CODE REVIEW

File: src/AutoDeploy/ServiceInstaller/App/Program.cs [hunk 1]
```diff
@@ -107,39 +107,75 @@ public static void RunIt(string appName)
...
+                return 0;
             }
 
-            public static bool GenerateUnzipCommand(string applicationName)
+            public static bool ExtractZip(string applicationName)
             {
                 try
                 {
                     Console.WriteLine("Unzipping");
                     string zipPath = @"C:\Upgrade\AutoDeploy\" + applicationName + @"\" + applicationName + ".zip";
```
Comment (jsnmc) [inline src/AutoDeploy/ServiceInstaller/App/Program.cs]: This reminds me that I would like to go through end-to-end and remove all the hard-coded C:\Upgrade paths and move it to relative pathing or make the install root a config option;

**Comment by jsnmc** [comment]
Looks good.

**Comment by jsnmc** [comment]
Looks good - Sorry, hit the close button by accident :(

File: src/AutoDeploy/ServiceInstaller/App/Program.cs (hunk 1 above)
Comment (jakebrown58) [inline src/AutoDeploy/ServiceInstaller/App/Program.cs]: I'm not sure what we would gain.  It's simple and reliable that we require it to be placed at C:\Upgrade - and nobody has complained about that, and keeps us from tracking the working directory while we spawn different processes.  It don't see the hard-paths as hurting the quality of the product at this point.

**Comment by adampaulberg** [comment]
:+1:

File: src/AutoDeploy/ServiceInstaller/App/Program.cs (hunk 1 above)
Comment (jsnmc) [inline src/AutoDeploy/ServiceInstaller/App/Program.cs]: If the tool is strictly internal use, than I agree with you.  If we are going to ship to customers, they might not like a hard-coded install dir - no big deal.

**Pull request merged**